Libraries/MLXVLM/MediaProcessing.swift

```diff
@@ -199,4 +243,96 @@ public enum MediaProcessing {
 
         return ciImages
     }
+
+    static public func asProcessedSequence(_ asset: AVAsset, samplesPerSecond: Int, frameProcessing: (VideoFrame) throws -> VideoFrame = {$0}) async throws -> ProcessedFrames {
+        return try await asProcessedSequence(asset, maxFrames: Int.max, targetFPS: {_ in Double(samplesPerSecond)}, frameProcessing: frameProcessing)
+    }
+
+    static public func asProcessedSequence(_ asset: AVAsset, maxFrames: Int, targetFPS: (CMTime) -> Double, frameProcessing: (VideoFrame) throws -> VideoFrame = {$0}) async throws -> ProcessedFrames {
+        // Use AVAssetImageGenerator to extract frames
+        let generator = AVAssetImageGenerator(asset: asset)
+        generator.appliesPreferredTrackTransform = true
+        generator.requestedTimeToleranceBefore = .zero
+        generator.requestedTimeToleranceAfter = .zero
+
+        guard let duration = try? await asset.load(.duration) else {
+            throw NSError(
+                domain: "MediaProcessing", code: -1,
+                userInfo: [NSLocalizedDescriptionKey: "Failed to load the asset's duration"])
+        }
+        let fps = targetFPS(duration)
+        // Note: the round was not present in `asCIImageSequence`, so we may now be passing 1 more frame to Qwen depending on video duration.
```

Comment:
As noted in the comment, this may result in an additional frame being extracted for users of the previous method (only Qwen VL). I don't think this would be a big deal, so we can just remove the comment.

```suggestion
```